internal/js-dependency-scanner/src/main/java/org/wso2/security/tools/scanner/dependency/js/preprocessor/PreProcessor.java

```diff
@@ -0,0 +1,171 @@
+     * Entry point to start pre-processing task.
+     * Following tasks are sub task of this method.
+     * Parse configuration
+     * Download product packs
+     * Download package.json files
+     * Unzip product directory
+     *
+     * @return productFileMapper which holds the values each product's path where it is downloaded.
+     */
+    public HashMap<String, String> startPreprocessing() {
+
+        try {
+            List<String> productList;
+            //Parse config details.
+            //Get supported product list from config file
+            productList = ConfigParser.parseProductList();
+            List<Product> supportedProductList = new ArrayList<>();
+            //Get product details from respective product configuration files.
+            for (String product : productList) {
+                supportedProductList.add(ConfigParser.parseProductConfiguration(product));
+            }
+
+            //Create Root Directory for products if not exists
+            productRootDirectory = new File(JSScannerConstants.PRODUCT_HOME);
+            createDirectory(productRootDirectory);
+            log.info("[JS_SEC_DAILY_SCAN] Start downloading product packs");
+            //download resources
+            downloadResources(supportedProductList);
+            Arrays.fill(CommonApiInvoker.getGitToken(),
+                    JSScannerConstants.RANDOM_STRING.charAt(ConfigParser.getRandomNumber()));
+        } catch (ApiInvokerException | DownloaderException | ConfigParserException e) {
+            log.error("Error occurred while downloading product pack.", e);
+        }
+        return productFileMapper;
+    }
+
+    /**
+     * Download resources. It can be either product pack or package.json files.
+     * Product pack can be downloaded from any resource. Currently it supports GitHub.
+     * Package.json files can be downloaded from github repo as raw materials.
+     *
+     * @param productDtoList Product list
+     * @throws DownloaderException exception occurred while downloading files.
+     */
+    private void downloadResources(List<Product> productDtoList) throws DownloaderException,
+            ApiInvokerException, ConfigParserException {
+        ConfigParser.parseGitAccessToken();
+        productFileMapper = new HashMap<>();
+        String filename;
+        for (Product productDto : productDtoList) {
+            //create directory for current product
+            File currentProductDir = new File(productRootDirectory.getAbsolutePath() + File.separator +
+                    productDto.getProductRepoName());
+            createDirectory(currentProductDir);
+
+            List<String> zipFileList = new ArrayList<>();
+            //download from git repository
+            if (productDto.getInputSourceType().equals(JSScannerConstants.GIT)) {
+                ResourceDownloader gitDownloader = new GitDownloader();
+                startDownload(productDto, currentProductDir, zipFileList, gitDownloader);
```

Comment:
It is more readable if you return the zip file list from the startDownload method and then assign it to 'zipFileList' variable, rather than sending an empty list into the method as an argument and then populating that list inside the method.